lib/logmsg/tests/test_tags.c

```diff
@@ -122,10 +110,10 @@ test_msg_tags(void)
             log_msg_set_tag_by_name(msg, name);
           else
             log_msg_clear_tag_by_name(msg, name);
-          test_msg("%s tag %d %s\n", set ? "Setting" : "Clearing", i, name);
+          cr_log_info("%s tag %d %s\n", set ? "Setting" : "Clearing", i, name);
 
-          if (set ^ log_msg_is_tag_by_id(msg, i))
-            test_fail("Tag is %sset now (by id) %d\n", set ? "not " : "", i);
+          if(set ^ log_msg_is_tag_by_id(msg, i))
+            cr_log_info("Tag is %sset now (by id) %d\n", set ? "not " : "", i);
```

Comment:
This has to be an assert.

Reply:
I tried to change it to `cr_assert_null(set ^ log_msg_is_tag_by_id(msg, i), "Tag is %sset now (by id) %d\n", set ? "not " : "", i)`, but it fails. Similar for the rest three, i.e wherever there are bitwise operations this is failing. Please help.

Reply:
I'll have a look at this tomorrow. Have you tried `cr_assert_not`?

Reply:
Yes, I tried `cr_assert_not`, it also fails.

Reply:
@MrAnno Can you please give me any lead, so that I can solve this. Thank you.

Reply:
This is a bug in the old test case.
Previously, `test_tags` was executed before `test_msg_tags`, sequentially.
Now that you switched to Criterion, tests run in parallel.
Unfortunately, `test_tags` has some global side-effect that made this assertion succeed.

This side effect has to be found out and fixed.

Reply:
Okay I think I got it. The reversed order of the for loop does not really make sense here:

```diff
  LogMessage *msg = log_msg_new_empty();
  for (set = 1; set != -1; set--)
    {
      for (i = NUM_TAGS; i > -1; i--)
        {
          name = get_tag_by_id(i);
+         LogTagId id = log_tags_get_by_name(name);

          if (set)
            log_msg_set_tag_by_name(msg, name);
          else
            log_msg_clear_tag_by_name(msg, name);
+         cr_log_info("%s tag %d %s\n", set ? "Setting" : "Clearing", id, name);

-         if (set ^ log_msg_is_tag_by_id(msg, i))
-           test_fail("Tag is %sset now (by id) %d\n", set ? "not " : "", i);
+        cr_assert_not(set ^ log_msg_is_tag_by_id(msg, id), "Tag is %sset now (by id) %d", set ? "not " : "", id);
```

Or another version:
```diff
-for (i = NUM_TAGS; i > -1; i--)
+for (i = 0; i < NUM_TAGS; i++)
```

Reply:
Okay, Got it. Thanks :-)